### PR TITLE
mempool: Decouple mining-specific logic.

### PR DESCRIPTION
**Pull request by davecgh**
This splits the mining-specific code that deals with sorting blocks according to the number of votes available for it in the `mempool` into `mining.go`.

This is being done for a couple of reasons:

1) Mining-specific code does not belong directly in the `mempool`
2) It is required to be able to split the `mempool` into a separate package which an upcoming upstream sync does

In order to accomplish this, a new function named `VotesForBlocks` has been exposed on the `mempool` and the `SortParentsByVotes` code has been moved to `mining.go` and modified to no longer return errors.  The reasoning for this is the same as the recent change to `VoteHashesForBlock`.  That is to say it is not the responsibility of a sorting function to dictate caller-specific block eligibility logic.

### CODE REVIEW

**Comment by jcvernaleo** [comment]
The code reads correct.  Needs a bunch of testing before we should okay it though.

File: mining.go [hunk 1]
```diff
@@ -279,6 +280,107 @@ func containsTxIns(txs []*dcrutil.Tx, tx *dcrutil.Tx) bool {
...
+	// mempool and filter out any blocks that do not have the minimum
+	// required number of votes.
+	minVotesRequired := (params.TicketsPerBlock / 2) + 1
+	voteMetadata := mp.VotesForBlocks(blocks)
+	filtered := make([]*blockWithNumVotes, 0, lenBlocks)
+	for i := range blocks {
+		numVotes := uint16(len(voteMetadata[i]))
+		if numVotes >= minVotesRequired {
+			filtered = append(filtered, &blockWithNumVotes{
+				blocks[i], numVotes})
```
Comment (davecgh) [inline mining.go]: I'm not a fan in general, but the code was already that way, so I was attempting to change as little as possible to decouple it and correct the things that didn't make sense.

I will happily change it.

Comment (jrick) [inline mining.go]: it's fine, and personally I prefer the unkeyed version as it's harder to get it wrong by missing fields.

**Comment by jolan** [comment]
testnet mining tACK. Have mined two dozen blocks with this so far; no problems encountered.

File: mining.go (hunk 1 above)
Comment (davecgh) [inline mining.go]: Updated.

**Comment by alexlyp** [comment]
OK

**Pull request merged**